src/main/java/com/commercetools/sync/products/utils/ProductVariantUpdateActionUtils.java

```diff
@@ -117,39 +119,114 @@
     }
 
     /**
-     * Compares the images of a {@link ProductVariantDraft} and a {@link ProductVariant}.
-     * TODO: Add JavaDoc..
+     * Compares the {@link List} of {@link Image}s of a {@link ProductVariantDraft} and a {@link ProductVariant} and
+     * returns a {@link List} of {@link UpdateAction}&lt;{@link Product}&gt;. If both the {@link ProductVariantDraft}
+     * and the {@link ProductVariant} have identical list of images, then no update action is needed and hence an
+     * empty {@link List} is returned.
      *
-     * @param oldProductVariant TODO
-     * @param newProductVariant TODO
-     * @return TODO
+     * @param oldProductVariant the {@link ProductVariant} which should be updated.
+     * @param newProductVariant the {@link ProductVariantDraft} where we get the new list of images.
+     * @return a list that contains all the update actions needed, otherwise an empty list if no update actions are
+     *         needed.
      */
     @Nonnull
     public static List<UpdateAction<Product>> buildProductVariantImagesUpdateActions(
         @Nonnull final ProductVariant oldProductVariant,
         @Nonnull final ProductVariantDraft newProductVariant) {
         final List<UpdateAction<Product>> updateActions = new ArrayList<>();
+        final Integer oldProductVariantId = oldProductVariant.getId();
         final List<Image> oldProductVariantImages = oldProductVariant.getImages();
         final List<Image> newProductVariantImages = newProductVariant.getImages();
 
-        final Map<String, Image> oldUrlToImageMap = oldProductVariantImages != null
-            ? oldProductVariantImages.stream()
-                                     .collect(Collectors.toMap(Image::getUrl, image -> image)) : Collections.emptyMap();
-        final Map<String, Image> newUrlToImageMap = newProductVariantImages != null
-            ? newProductVariantImages.stream()
-                                     .collect(Collectors.toMap(Image::getUrl, image -> image)) : Collections.emptyMap();
+        // this implementation is quite straight forward and might be slow on large arrays,
+        // because of quadratic algorithms of remove/add images synchronization.
+        // Unfortunately, there is not easy solution to synchronize 2 ordered lists
+        // having only add/remove/moveToPos actions.
+        // This solution should be re-optimized in the next releases to avoid O(N^2) for large lists.
+
```

Comment:
i put my comments here, pls read them and make your notes (if necessary)

Reply:
I reworded the JavaDoc a bit and added a GH ISSUE TODO: https://github.com/commercetools/commercetools-sync-java/issues/133